source/TimePicker.js

```diff
@@ -53,14 +53,73 @@
 		published: {
 			/**
 			* The meridiem text to display if [meridiemEnable]{@link moon.TimePicker#meridiemEnable}
-			* is `true`. The first item is used if the `hour` is less than `11`; otherwise, the
+			* is `true`. The first item is used if the `hour` is less than `12`; otherwise, the
 			* second is used.
 			*
 			* @type {String[]}
-			* @default ['AM','PM']
+			* @default [{name: 'AM', start: '00:00', end: '11:59'}, {name: 'PM', start: '12:00', end: '23:59'}]
 			* @public
 			*/
-			meridiems: ['AM','PM']
+			meridiems: [{name: 'AM', start: '00:00', end: '11:59'}, {name: 'PM', start: '12:00', end: '23:59'}]
+		},
+
+		/**
+		* @private
+		*/
+		initILib: function() {
+			// Get localized meridiem values
+			var fmtParams = {
+				template: 'a',
+				useNative: false,
+				timezone: 'local'
+			};
+
+			if (this.locale) fmtParams.locale = this.locale;
+
+			var merFormatter = new ilib.DateFmt(fmtParams);	
+			this.meridiems = merFormatter.getMeridiemsRange(fmtParams);
+		},
+
+		/**
+		* @private
+		*/
+		create: function() {
+			this.initILib();
+			this.max = this.meridiems.length - 1;
+			this.inherited(arguments);	
+		},
+
+		/**
+		* Return index of meridiems array for given hour
+		*
+		* @param  {Number} hour - hour between 0 to 23
+		* @param  {Number} minute - minute between 0 to 59
+		* @return {Number} 		- index of this.meridiems
+		*
+		* @public
+		*/
+		getMeridiemIndex: function (hour, minute) {
+			var meridiems = this.meridiems,
+				start, end, time;
+			for (var i = 0; i < meridiems.length; i++) {
+				start = parseInt(meridiems[i]['start'].substring(0,2) + meridiems[i]['start'].substring(3,5), 10);
+				end = parseInt(meridiems[i]['end'].substring(0,2) + meridiems[i]['end'].substring(3,5), 10);
+				time = hour * 100 + minute;
+					
+				if ( start <= time && time <= end) return i; 
+			}
+		},
+
+		/**
+		* Set meridiem value based on given hour and minute
+		*
+		* @param  {Number} hour - hour between 0 to 23
+		* @param  {Number} minute - minute between 0 to 59
+		*
+		* @public
+		*/
+		setValueByTime: function (hour, minute) {
+			this.value = this.getMeridiemIndex(hour, minute);
```

Comment:
Should this call the setter instead, i.e. `this.set('value', this.getMeridiemIndex(hour, minute)`? Also, I'm wondering if we can fold `getMeridiemIndex` into this function as I don't think it needs to be a separate, public function - it is logic internal to `MeridiemPicker` that only it should know about.